spec/unit/functions/epp_spec.rb

```diff
@@ -18,8 +18,13 @@
     end
 
     it "gets error accessing a variable that is malformed" do
-      expect { eval_template("<%= $kryptonite::USER %>")}.to raise_error(
-        /Illegal variable name, The given name 'kryptonite::USER' does not conform to the naming rule/)
+      expect { eval_template("<%= $kryptonite::bbbbbbbbbbbb::cccccccc::ddd::USER %>")}.to raise_error(
+        /Illegal variable name, The given name 'kryptonite::bbbbbbbbbbbb::cccccccc::ddd::USER' does not conform to the naming rule/)
```

Comment:
The problem was that the sample expression took up to a minute to evaluate. This test doesn't verify that it no longer does (green both before and after the fix even if it takes a while). I'd like to see a test that verifies that the evaluation completes within an acceptable time frame. On my machine, the new expression (both yours and mine) evaluates within 2 usecs so acceptable in this case could be 1 msec.